### PR TITLE
Ticket 3745 - Create IOC for Keyence LK-G laser positioning sensor

### PR DESCRIPTION
**Pull request by aaron-long**
Created unit tests for the Keyence LK-G laser positioning sensor.

### CODE REVIEW

**Comment by LiamPanchaud** [comment]
I didn't see any tests that test the functionality of records that modift both heads (e.g RESET:OUTPUT:BOTH:SP). These would be good to add to ensure there isn't any unexpected beahaviour when modifying both heads.

File: tests/keylkg.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        self.ca.assert_that_pv_is("MODE", expected_value, timeout=2)
+
+    def test_GIVEN_running_ioc_WHEN_change_to_normal_mode_THEN_mode_changed(self):
+        expected_value = "MEASURE"
+        self.ca.set_pv_value("MODE:SP", expected_value)
+
+        self.ca.assert_that_pv_is("MODE", expected_value, timeout=2)
+
+    def test_GIVE_running_ioc_WHEN_set_output1_offset_THEN_output1_offset_updated(self):
+        expected_value = 0.123
```
[the remark this answers is not in the document]
Comment (aaron-long) [reply tests/keylkg.py]: Added additional tests across a range of values and limits.

**Comment by aaron-long** [comment]
> 
> 
> I didn't see any tests that test the functionality of records that modift both heads (e.g RESET:OUTPUT:BOTH:SP). These would be good to add to ensure there isn't any unexpected beahaviour when modifying both heads.

Updated tests to include reset functions.

File: tests/keylkg.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        self.ca.set_pv_value("OFFSET:OUTPUT:1:SP", expected_value)
+
+        self.ca.assert_that_pv_is("OFFSET:OUTPUT:1", expected_value, timeout=2)
+
+    @parameterized.expand([('exceeds low limit', -100.0000), ('exceeds high limit', 100.000)])
+    def test_GIVE_running_ioc_WHEN_set_output1_offset_outside_of_limits_THEN_output1_offset_within_limits(self, _, mock_offset):
+        expected_value = mock_offset
+        self.ca.set_pv_value("OFFSET:OUTPUT:1:SP", expected_value)
+
+        self.ca.assert_that_pv_is_an_integer_between("OFFSET:OUTPUT:1", -99.9999, 99.9999)
```
[the remark this answers is not in the document]
Comment (aaron-long) [reply tests/keylkg.py]: That's strange, I already implemented `assert_that_pv_is_within_range` into channel access utils to be used for this test! I've updated it [test].

**Pull request merged**